File: server/models/user.model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require("../config/db");

const Users = sequelize.define(
    "users",
    {
        id: {
            type: DataTypes.BIGINT,
            primaryKey: true,
            autoIncrement: true,
        },
        name: {
            type: DataTypes.STRING,
            allowNull: false
        },
        email: {
            type: DataTypes.STRING,
            allowNull: false,
            unique: true
        },
        password: {
            type: DataTypes.TEXT,
            allowNull: false
        },
        role: {
            type: DataTypes.ENUM('PORTAL_ADMIN', 'OWNER', 'STAFF'),
            defaultValue: 'OWNER'
        },
        status: {
            type: DataTypes.BOOLEAN,
            defaultValue: true
        },
        suspended_reason: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        created_at: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW
        },
        profile_picture: {
            type: DataTypes.TEXT('long'),
            allowNull: true
        },
        google_id: {
            type: DataTypes.STRING,
            allowNull: true,
            unique: true
        },
        avatar: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        auth_provider: {
            type: DataTypes.ENUM('local', 'google'),
            defaultValue: 'local'
        }
    },
    {
        timestamps: false,
        tableName: "users",
        indexes: [
            { fields: ["status"] }
        ]
    }
);

module.exports = Users;

File: my_booking_application/lib/app/data/models/user_model.dart
class UserModel {
  final int? id;
  final String? name;
  final String? email;
  final String? role;
  final bool? status;
  final String? profilePicture;
  final String? token;
  final int? businessId;

  UserModel({
    this.id,
    this.name,
    this.email,
    this.role,
    this.status,
    this.profilePicture,
    this.token,
    this.businessId,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) => UserModel(
        id: json['id'],
        name: json['name'],
        email: json['email'],
        role: json['role'],
        status: json['status'],
        profilePicture: json['profile_picture'] ?? json['avatar'],
        token: json['token'],
        businessId: json['business_id'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'email': email,
        'role': role,
        'status': status,
        'profile_picture': profilePicture,
        'token': token,
        'business_id': businessId,
      };
}
